src/Model/Subscription/SubscriptionBuilder.php

```diff
@@ -127,25 +130,31 @@ public function withDiscount(Discount $discount): self
 
     public function build(): Subscription
     {
-        $subscription = (new Subscription($this->get('id')))
-            ->setStartDate($this->get('startDate'))
-            ->setPrice($this->get('price'))
-            ->setStatus($this->get('status'))
-            ->setCustomer($this->get('customer'))
-            ->setPaymentMethod($this->get('paymentMethod'))
-            ->setBalance($this->get('balance'));
-
-        $addOns = $this->get('addOns') ?? [];
+        $subscription = (new Subscription($this->getId()))
+            ->setStartDate($this->data['startDate'] ?? new DateTime('now', new DateTimeZone('UTC')))
```

Comment:
Shouldn't this do the `beginsImmediately()` if `startDate` is not set?

Reply:
Good catch.